### PR TITLE
feat: [CDS-91915]: Add Knowledge base how bidirectional sync prevent github api limit

### PR DESCRIPTION
**Pull request by krishi0408**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## API rate limits, what causes them?
+
+Github limits the number of REST API requests that you can make within a specific amout of time. Sometimes, the number is extrememly high and stops remote entitites from working. 
+
+## Solving GitHub API Limits with Bidirectional Sync
+
+In bidirectioal sync, changes made on either the Git repository or Harness are automatically synchronized. Updates to the entity on the platform are determined by the received event payload.Therefore, it is required to enable bidirectional sync congiguration before the remote entities are created. 
+
+Read more on how to enable [Biderectional Sync](/docs/platform/git-experience/gitexp-bidir-sync-setup.md).
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:17]:
```suggestion
Read more on how to enable [Biderectional Sync](/docs/platform/git-experience/gitexp-bidir-sync-setup/).
```

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+description: KB - How Bidirectional sync prevent Github API Limit issue
+title: Preventing GitHub API Rate Limits with Bidirectional Sync
+---
+# Introduction
+
+This knowledge base article discusses how Bidirectional Sync can help prevent reaching the GitHub API limit issue.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:7]:
```suggestion
This knowledge base article discusses how bidirectional sync can help prevent reaching the GitHub API limit issue.
```

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## API rate limits, what causes them?
+
+Github limits the number of REST API requests that you can make within a specific amout of time. Sometimes, the number is extrememly high and stops remote entitites from working.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:11]:
```suggestion
GitHub limits the number of REST API requests that you can make within a specific amount of time. Sometimes, the number is high and stops remote entities from working. 
```

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+This knowledge base article discusses how Bidirectional Sync can help prevent reaching the GitHub API limit issue.
+
+## API rate limits, what causes them?
+
+Github limits the number of REST API requests that you can make within a specific amout of time. Sometimes, the number is extrememly high and stops remote entitites from working. 
+
+## Solving GitHub API Limits with Bidirectional Sync
+
+In bidirectioal sync, changes made on either the Git repository or Harness are automatically synchronized. Updates to the entity on the platform are determined by the received event payload.Therefore, it is required to enable bidirectional sync congiguration before the remote entities are created.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:15]:
```suggestion
In bidirectional sync, changes made on either the Git repository or Harness are automatically synchronized. Updates to the entity on the platform are determined by the received event payload. Therefore, it is required to enable bidirectional sync configuration before the remote entities are created. 
```

**Review by roshnisarangadharan — APPROVED** [review]
Fixed typos and links. Approved.

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+description: KB - How Bidirectional sync prevent Github API Limit issue
+title: Preventing GitHub API Rate Limits with Bidirectional Sync
+---
+# Introduction
+
+This knowledge base article discusses how bidirectional sync can help prevent reaching the GitHub API limit issue.
```
Comment (adivishy1) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:7]: We dont need to specify a git provider, we can just tell Git rate limit issue

Comment (adivishy1) [reply kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:7]: you can change this everywhere GitHub is mentioned

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+## API rate limits, what causes them?
+
+Git limits the number of REST API requests that you can make within a specific amount of time. Sometime, the number of API request at a specific period of time is high and it stops the remote entitites from working.
+
+## Solving GitHub API Limits with Bidirectional Sync
+
+In bidirectional sync, changes made on either the Git repository or Harness are automatically synchronized. With bidirectional sync, files are cached at Harness's end, creating a copy of Git. Instead of making frequent requests to the Git API for every operation, the system can rely on the cached data for all tasks, thus reducing overall API usage.
+
+In bidirectional sync, the source of truth is Harness's cached data. Therefore, instead of making API calls to Git every time, bidirectional sync fetches the data from Harness Cache, thus limiting the risk of reaching API limits.
```
Comment (SushrutHarness) [inline kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md:17]: You've said the same thing twice in line 15 and 17, I think one of these might be leftovers from an edit.

**Review by SushrutHarness — CHANGES_REQUESTED** [review]
I know you've done the work already, but I'm wondering if this should just be added to the bidirectional sync documentation as opposed to being its own KB article.

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md (hunk 6 above)
Comment (krishi0408) [reply kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md]: Yup didn't noticed, thanks for pointing @SushrutHarness. I have removed the second para

**Comment by krishi0408** [comment]
> I know you've done the work already, but I'm wondering if this should just be added to the bidirectional sync documentation as opposed to being its own KB article.

Hey @SushrutHarness yes we thought about that initially but then PMs and dev decided to have a separate KB on how bidirectional sync prevent hitting API rate limits and attach the doc there.

File: kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+---
+description: KB - How Bidirectional sync prevent Git API Limit issue
+title: Preventing Git API Rate Limits with Bidirectional Sync
+---
+# Introduction
+
+This knowledge base article discusses how bidirectional sync can help prevent reaching the Git API limit issue.
```
[the remark this answers is not in the document]
Comment (krishi0408) [reply kb/continuous-delivery/articles/biderectional-sync-prevent-github-api-limit.md]: Yup added rate, thanks for noticing. Appreciate it.

**Pull request merged**